cassandra-analytics-common/src/main/java/org/apache/cassandra/spark/data/CqlTable.java

```diff
@@ -275,6 +275,37 @@ public boolean containsUdt(String fieldName)
         return columnsWithUdts.contains(fieldName);
     }
 
+    /**
+     * If a field is of type Tuple, or Tuple inside a frozen type, this function finds CqlTuple associated with that field
+     * @param fieldName name of the field
+     * @return CqlTuple associated with the requested field, returns null otherwise
+     */
+    public CqlField.CqlTuple findTuple(String fieldName)
+    {
+        CqlField field = fieldsMap.get(fieldName);
+        if (field == null)
+        {
+            return null;
+        }
+
+        CqlField.CqlType current = field.type();
+        while (true)
+        {
+            if (current instanceof CqlField.CqlTuple)
+            {
+                return (CqlField.CqlTuple) current;
+            }
+
+            if (current instanceof CqlField.CqlFrozen)
+            {
+                CqlField.CqlFrozen frozen = (CqlField.CqlFrozen) current;
+                current = frozen.inner();
+                continue;
+            }
+            return null;
+        }
+    }
```

Comment:
To clarify, this implementation does not cover the case where tuple is inside collections. 

For example, `tupleList list<tuple<text, text>>`, when entering the loop, it will return `null` immediately as the outer data type is a list, not matching `CqlTuple` or `CqlFrozen`. This is problematic. 

To make things more clear, I added the test to demonstrate. See https://github.com/yifan-c/cassandra-analytics/commit/3b05c189831a164394eb395a3bf9d0b9d2f696d4